ast/desugar/Desugar.cc

```diff
@@ -825,8 +825,15 @@ ExpressionPtr node2TreeImpl(DesugarContext dctx, unique_ptr<parser::Node> what)
                 } else {
                     int numPosArgs = send->args.size();
                     if (numPosArgs > 0) {
-                        // Deconstruct the kwargs hash in the last argument if it's present.
-                        if (auto *hash = parser::cast_node<parser::Hash>(send->args.back().get())) {
+                        // The keyword arguments hash can be the last argument if there is no block
+                        // or second to last argument otherwise
+                        int kwargsHashIndex = send->args.size() - 1;
+                        if (!parser::isa_node<parser::Hash>(send->args[kwargsHashIndex].get())) {
```

Comment:
If I'm understanding this correctly, the logic implemented here is:
- if the last arg is a hash, use that as the kwargsHashIndex
- otherwise, check if the second last arg is a hash, and if it is, use that

The comment also mentions the second last arg only fills this role if the last arg is a block, but we don't check that here. Could that be an issue here?

Reply:
I don't believe that might be an issue. My goal was to fix how keyword arguments are parsed without touching the block parsing, because I believe it works correctly now. Tests also passed, so I think we are good!

I also don't do a check for the block argument, because it's done later in the same function: https://github.com/sorbet/sorbet/blob/ce70710668aaadd9aee1ba676bdb6d856f12c383/ast/desugar/Desugar.cc#L872-L888

Please let me know, if you want me to update my comment to point to the block processing code!